### PR TITLE
DLS-9905: Suggest Deregistration for Large Producers

### PR DESCRIPTION
**Pull request by drewmiley**

### CODE REVIEW

**Comment by drewmiley** [comment]
Screenshot 2024-01-30 at 11 35 10

File: test/navigation/NavigatorForChangeActivitySpec.scala [hunk 1]
```diff
@@ -145,10 +178,29 @@ class NavigatorForChangeActivitySpec extends SpecBase {
         result mustBe routes.ImportsController.onPageLoad(NormalMode)
       }
 
-      "Should navigate to Check Your Answers page when no is selected in CheckMode" in {
+      "Should navigate to Check Your Answers page when no is selected in CheckMode and deregistration not suggested" in {
         val result = navigateFromContractPackingPage(value = false, CheckMode)
         result mustBe routes.ChangeActivityCYAController.onPageLoad
       }
+
+      "should navigate to suggest registration in CheckMode when Amount Produced Small, not TPP, not Contract Packing, not Importing" in {
```
Comment (GinnyThomas) [inline test/navigation/NavigatorForChangeActivitySpec.scala:186]: suggest registration should be suggest deregistration

File: test/navigation/NavigatorForChangeActivitySpec.scala [hunk 2]
```diff
@@ -145,10 +178,29 @@ class NavigatorForChangeActivitySpec extends SpecBase {
...
+        val result = navigator.nextPage(ContractPackingPage, CheckMode, emptyUserAnswersForChangeActivity
+          .set(AmountProducedPage, AmountProduced.Small).success.value
+          .set(ThirdPartyPackagersPage, false).success.value
+          .set(ContractPackingPage, false).success.value
+          .set(ImportsPage, false).success.value
+        )
+        result mustBe routes.SuggestDeregistrationController.onPageLoad
+      }
+
+      "should navigate to suggest registration in CheckMode when Amount Produced None, not Contract Packing, not Importing" in {
```
Comment (GinnyThomas) [inline test/navigation/NavigatorForChangeActivitySpec.scala:196]: same as 186

**Review by GinnyThomas — DISMISSED** [review]
LGTM.  Good refactoring

**Review by GinnyThomas — APPROVED** [review]

**Pull request merged**